src/ExpressionEvaluator/Core/Source/ResultProvider/Expansion/DynamicViewExpansion.cs

```diff
@@ -78,30 +78,27 @@ private DynamicViewExpansion(DkmClrValue proxyValue, Expansion proxyMembers)
         {
             if (InRange(startIndex, count, index))
             {
-                rows.Add(CreateDynamicViewRow(inspectionContext, Resources.DynamicView, parent, resultProvider.Formatter));
+                rows.Add(CreateDynamicViewRow(inspectionContext, Resources.DynamicView, parent, resultProvider.FullNameProvider));
             }
 
             index++;
         }
 
-        private EvalResultDataItem CreateDynamicViewRow(DkmInspectionContext inspectionContext, string name, EvalResultDataItem parent, Formatter formatter)
+        private EvalResult CreateDynamicViewRow(DkmInspectionContext inspectionContext, string name, EvalResultDataItem parent, IDkmClrFullNameProvider fullNameProvider)
         {
             var proxyTypeAndInfo = new TypeAndCustomInfo(_proxyValue.Type);
             var isRootExpression = parent == null;
-            Debug.Assert(isRootExpression != (name == Resources.DynamicView));
             var fullName = isRootExpression ? name : parent.ChildFullNamePrefix;
-            var sawInvalidIdentifier = false;
             var childFullNamePrefix = (fullName == null) ?
                 null :
-                formatter.GetObjectCreationExpression(formatter.GetTypeName(proxyTypeAndInfo, escapeKeywordIdentifiers: true, sawInvalidIdentifier: out sawInvalidIdentifier), fullName);
-            Debug.Assert(!sawInvalidIdentifier); // Expected type name is "Microsoft.CSharp.RuntimeBinder.DynamicMetaObjectProviderDebugView".
+                fullNameProvider.GetClrObjectCreationExpression(inspectionContext, proxyTypeAndInfo.ClrType, proxyTypeAndInfo.Info, fullName);
```

Comment:
If the dynamic view proxy type has an invalid name, GetClrObjectCreationExpression will return null.